src/Sync.ts

```diff
@@ -0,0 +1,200 @@
+import { ComponentEvent } from "@egjs/component";
+import Flicking, { EVENTS, Plugin } from "@egjs/flicking";
+
+import { SYNC } from "./const";
+import { addClass, removeClass } from "./utils";
+
+interface SyncOptions {
+  type: "camera" | "index";
+  synchronizedFlickings: SychronisableFlickingOptions[];
+}
+
+interface SychronisableFlickingOptions {
+  flicking: Flicking;
+  isClickable?: boolean;
+  isSlidable?: boolean;
+  activeClass?: string;
+}
+
+/**
+ * Plugin for synchronizing multiple flickings 
+ * @ko 다양한 형태로 Flicking들이 같이 움직이게 할 수 있습니다.
+ * @memberof Flicking.Plugins
+ */
+class Sync implements Plugin {
+  /* Internal Values */
+  private _flicking: Flicking | null = null;
+
+  /* Options */
+  private _type: SyncOptions["type"];
+  private _synchronizedFlickings: SyncOptions["synchronizedFlickings"];
+
+  public get type() { return this._type; }
+  public get synchronizedFlickings() { return this._synchronizedFlickings; }
+
+  public set type(val: SyncOptions["type"]) {
```

Comment:
유사해보이는 코드인데요. 제네릭을 써서 개선해볼 수 있을까요?
This code looks similar. Can we improve it by using generics?
```typescript
public set type(val: SyncOptions["type"]) {
    this.restore<SyncOptions["type"]>(val, "_type");
}

public set synchronizedFlickings(val: SyncOptions["synchronizedFlickings"]) {
    this.restore<SyncOptions["synchronizedFlickings"]>(val, "_synchronizedFlickings");
}

private restore<T>(val:T, type:"_type"|"_synchronizedFlickings"){
    this._removeEvents(this_synchronizedFlickings);
    this[type] = val;
    this._addEvents(this_synchronizedFlickings);
}
```

Reply:
개인적으로 오브젝트의 키를 함수 인자로 넘기는것을 선호하지 않는데, 해당 코드의 목적이 이벤트로 인한 예상하지 못한 동작을 방지하는 것이라면 전후로 이벤트를 떼고 붙이는 고차 함수로 작성하는 것은 어떨까요?
Personally, I do not prefer to pass the object's key as a function argument. If the purpose of the code is to prevent unexpected behavior from event, how about writing it as a higher-order function that separates and attaches events before and after?

```
public set type(val: SyncOptions["type"]) {
    this.preventEvent(() => {
        this._type = val;
    });
}

public set synchronizedFlickings(val: SyncOptions["synchronizedFlickings"]) {
    this.preventEvent(() => {
        this._synchronizedFlickings = val;
    })
}

private preventEvent(fn: () => void) {
    this._removeEvents(this_synchronizedFlickings);
    fn();
    this._addEvents(this_synchronizedFlickings);
}
```

Reply:
https://github.com/granberia/egjs-flicking-plugins/blob/24bb65d256fcfee80d3bb17fe2341cb7a12dfdc2/src/Sync.ts#L177-L188

고차 함수를 사용하는 것으로 기존의 `_synchronizeByIndex` 또한 개선이 가능하다고 판단하여 해당 구현을 사용하였습니다.
Decided that the existing `_synchronizeByIndex` could also be improved, so that implementation was used.

```
    this.preventEvent(() => {
      synchronizedFlickingOptions.forEach((synchronizedFlickingOption) => {
        // calculate new target flicking position with active flicking size and target flicking size
        const targetLastPanel = synchronizedFlickingOption.flicking.panels[synchronizedFlickingOption.flicking.panels.length - 1];
        const targetPos = activePanel.position / (lastPanel.position + (lastPanel.size / 2)) * (targetLastPanel.position + (targetLastPanel.size / 2));
  
        synchronizedFlickingOption.flicking.control.moveToPosition(targetPos, 500);
        if (synchronizedFlickingOption.activeClass) {
          this._updateClass(synchronizedFlickingOption, targetPos);
        }
      });
    });
```